packages/x-date-pickers/src/MultiSectionDigitalClock/MultiSectionDigitalClock.tsx

```diff
@@ -143,10 +145,16 @@ export const MultiSectionDigitalClock = React.forwardRef(function MultiSectionDi
 
   const views = React.useMemo<readonly TimeViewWithMeridiem[]>(() => {
     if (!ampm || !inViews.includes('hours')) {
-      return inViews;
+      return isRtl ? inViews.toReversed() : inViews;
     }
-    return inViews.includes('meridiem') ? inViews : [...inViews, 'meridiem'];
-  }, [ampm, inViews]);
+
+    if (!isRtl) {
+      return inViews.includes('meridiem') ? inViews : [...inViews, 'meridiem'];
+    }
+
+    const viewsWithoutMeridiem = inViews.filter((view) => view !== 'meridiem');
+    return viewsWithoutMeridiem.toReversed().concat('meridiem');
```

Comment:
We are playing with fire here regarding views. 🙈 
Making such structural `views` adjustments is tricky and this is the place where such change creates the most problems. 🙈 
The tabulation/views process does not work as expected in RTL mode.
Given a demo with default settings, the minutes view is skipped—selecting hours jumps the focus to meridiem and after selecting it the Picker closes.

Have you tried adjusting only the views intended for rendering?